src/main/java/core/basesyntax/MyHashMap.java

```diff
@@ -1,19 +1,93 @@
 package core.basesyntax;
 
 public class MyHashMap<K, V> implements MyMap<K, V> {
+    private static final int DEFAULT_INITIAL_CAPACITY = 16;
+    private static final float DEFAULT_LOAD_FACTOR = 0.75f;
+
+    private Node<K, V>[] table;
+    private int size;
+    private int threshold;
+    private final float loadFactor;
+
+    public MyHashMap() {
+        this(DEFAULT_INITIAL_CAPACITY, DEFAULT_LOAD_FACTOR);
+    }
+
+    @SuppressWarnings("unchecked")
+    public MyHashMap(int initialCapacity, float loadFactor) {
+        this.table = (Node<K, V>[]) new Node[initialCapacity];
+        this.loadFactor = loadFactor;
+        this.threshold = (int) (initialCapacity * loadFactor);
+    }
 
     @Override
     public void put(K key, V value) {
+        int hash = (key == null) ? 0 : Math.abs(key.hashCode());
+        int index = hash % table.length;
```

Comment:
```suggestion
        int hash = (key == null) ? 0 : Math.abs(key.hashCode());
        int index = hash % table.length;
```

create separate method for calculate index